### PR TITLE
Update interpretation of meas_map in Qiskit

### PR DESCRIPTION
**Pull request by SooluThomas**
### Summary

The interpretation of the meas_map has historically been:
- Any measurement of `qubit<i>` must be accompanied by a measurement of all qubits in `meas_map[i]`

The interpretation should now be

- Qubits in `meas_map[i]` may not be measured disjointly in time such that they overlap incompletely, eg., for `meas_map=[[0, 1]]` and duration 10. `measure(0)@t=10`, `measure(1)@t=10` is fine, but `measure(0)@t=10`, `measure(1)@t=11` is incorrect. `measure(0)@t=10`, `measure(1)@t=20` would be fine as they do not overlap
This must be reflected in both the circuit and pulse paths.

### Details and comments
- [x] **Part - 1 Fixing meas_map logic in measure of macro.py** - All this used to do was add extra acquire for even unused qubits. So, if I have meas map [[0, 1]], and zero is measured, 1 ALSO gets acquired. It doesn’t need to be. This PR fixes this issue.

- [x] **Part - 2 Change the way we validate schedules** - The meas map is interpreted incorrectly. If I acquire only 0 or only 1, theres no problem. If I acquire both, they MUST be at the same time, or else completely disjoint (finish acquiring 0 before starting to acquire 1)

Co-authored by: Lauren Capelluto (@lcapelluto)

### CODE REVIEW

File: qiskit/assembler/assemble_schedules.py [hunk 1]
```diff
@@ -232,18 +232,25 @@ def _validate_meas_map(instruction_map: Dict[Tuple[int, instructions.Acquire],
         QiskitError: If the instructions do not satisfy the measurement map.
     """
     meas_map_sets = [set(m) for m in meas_map]
-
-    # Check each acquisition time individually
-    for _, instrs in instruction_map.items():
```
Comment (lcapelluto) [inline qiskit/assembler/assemble_schedules.py:237]: There may be a more efficient way to do this. You might want to think about starting from scratch to get a clearer implementation -- the old one might be getting in your way.

Check out what `instruction_map` is giving you. Previously, the times were ignored. In `for _, instrs`, the `_` is a tuple which should contain the start time of the instruction, if the docstring is correct! So the work being done to create `time_mapping` has been done previously by the caller.

See what you can do by thinking through that -- try to come up with the lowest time complexity you think is possible. Think about what minimum information you need to decide if the schedule is invalid.

We can chat on webex, if you like, whenever you're ready.

**Review by lcapelluto — CHANGES_REQUESTED** [review]

File: qiskit/assembler/assemble_schedules.py [hunk 2]
```diff
@@ -232,18 +232,25 @@ def _validate_meas_map(instruction_map: Dict[Tuple[int, instructions.Acquire],
...
+    for meas_set in meas_map_sets:
+        time_mapping = defaultdict(list)
+        # Check each acquisition time individually
+        for time, instrs in instruction_map.items():  # time = (start_time, duration)
+            measured_qubits = set()
+            for inst in instrs:
+                time_mapping[inst.channel.index].extend(time)
+                measured_qubits.add(inst.channel.index)
+
+            if not measured_qubits.issubset(meas_set):
```
[the remark this answers is not in the document]
Comment (SooluThomas) [reply qiskit/assembler/assemble_schedules.py:254]: I've added a test for this and added a fix in https://github.com/Qiskit/qiskit-terra/pull/5971/commits/882a39fe9a525c3f793c48b0a350c5cc5dbfd414.

File: test/python/compiler/test_assembler.py [hunk 3]
```diff
@@ -1184,6 +1184,72 @@ def test_assemble_single_instruction(self):
...
+            Acquire(5, AcquireChannel(2), MemorySlot(2)),
+        )
+        schedule = schedule.append(
+            Acquire(5, AcquireChannel(3), MemorySlot(3)),
+        )
+        with self.assertRaises(QiskitError):
+            assemble(schedule,
+                     qubit_lo_freq=self.default_qubit_lo_freq,
+                     meas_lo_freq=self.default_meas_lo_freq,
+                     meas_map=[[0, 1, 2]])
```
Comment (lcapelluto) [inline test/python/compiler/test_assembler.py:1251]: Check that this is also fine?
```suggestion
                     meas_map=[[0, 1, 2], [3]])
```

File: qiskit/pulse/macros.py [hunk 4]
```diff
@@ -74,15 +74,16 @@ def measure(qubits: List[int],
...
             if qubit_mem_slots and isinstance(inst, instructions.Acquire):
                 if inst.channel.index in qubit_mem_slots:
                     mem_slot = channels.MemorySlot(qubit_mem_slots[inst.channel.index])
                 else:
                     mem_slot = channels.MemorySlot(unused_mem_slots.pop())
-                schedule = schedule.insert(time, instructions.Acquire(
-                    inst.duration, inst.channel, mem_slot=mem_slot))
-            elif qubit_mem_slots is None and isinstance(inst, instructions.Acquire):
-                schedule = schedule.insert(time, inst)
+                if inst.channel.index in qubits:
```
Comment (lcapelluto) [inline qiskit/pulse/macros.py:84]: Can remove this line

File: qiskit/pulse/macros.py [hunk 5]
```diff
@@ -74,15 +74,16 @@ def measure(qubits: List[int],
...
+                    schedule = schedule.insert(time, instructions.Acquire(
+                        inst.duration, inst.channel, mem_slot=mem_slot))
             # Measurement pulses should only be added if its qubit was measured by the user
             elif inst.channels[0].index in qubits:
```
Comment (lcapelluto) [inline qiskit/pulse/macros.py:88]: else

File: qiskit/assembler/assemble_schedules.py [hunk 6]
```diff
@@ -241,18 +241,30 @@ def _validate_meas_map(instruction_map: Dict[Tuple[int, instructions.Acquire],
...
-        measured_qubits = set()
-        for inst in instrs:
-            measured_qubits.add(inst.channel.index)
-
-        for meas_set in meas_map_sets:
-            intersection = measured_qubits.intersection(meas_set)
-            if intersection and intersection != meas_set:
-                raise QiskitError('Qubits to be acquired: {} do not satisfy required qubits '
-                                  'in measurement map: {}'.format(measured_qubits, meas_set))
+    # import ipdb; ipdb.set_trace();
```
Comment (lcapelluto) [inline qiskit/assembler/assemble_schedules.py:244]: Different cases to consider:

1. There are no overlapping acquisitions (No pair of keys `(time, duration)` with overlapping intervals) => no error
2. There are overlapping acquisitions and:
      - the two instructions acquire qubits which aren't in the same measurement group (qubits from first acquire: `[0, 1]`, qubits from second acquire: `[2, 3]`, measurement map: `[[0, 1], [2, 3]]`) => no error
      - at least one qubit is common to both acquire instructions (qubits from first acquire: `[0, 1, 2]`, qubits from second acquire: `[1]`) => error
      - the two instructions acquire different qubits but they share at least one measurement group (qubits from first acquire: `[0, 1, 2]`, qubits from second acquire: `[3, 4]`, measurement map: `[[0, 1], [2, 3], [4]]`) => error

**Review by lcapelluto — CHANGES_REQUESTED** [review]

File: test/python/compiler/test_assembler.py (hunk 3 above)
Comment (SooluThomas) [reply test/python/compiler/test_assembler.py:1251]: Fixed in https://github.com/Qiskit/qiskit-terra/pull/5971/commits/65a3fc296ab5ed9747ef3df78aaf1f82e7741f7c

File: qiskit/pulse/macros.py (hunk 5 above)
Comment (SooluThomas) [reply qiskit/pulse/macros.py]: fixed in https://github.com/Qiskit/qiskit-terra/pull/5971/commits/65a3fc296ab5ed9747ef3df78aaf1f82e7741f7c

File: qiskit/pulse/macros.py (hunk 4 above)
Comment (SooluThomas) [reply qiskit/pulse/macros.py]: Fixed in https://github.com/Qiskit/qiskit-terra/pull/5971/commits/65a3fc296ab5ed9747ef3df78aaf1f82e7741f7c

File: qiskit/pulse/macros.py [hunk 7]
```diff
@@ -81,11 +81,10 @@ def measure(qubits: List[int],
                     mem_slot = channels.MemorySlot(qubit_mem_slots[inst.channel.index])
                 else:
                     mem_slot = channels.MemorySlot(unused_mem_slots.pop())
-                if inst.channel.index in qubits:
-                    schedule = schedule.insert(time, instructions.Acquire(
-                        inst.duration, inst.channel, mem_slot=mem_slot))
+                schedule = schedule.insert(time, instructions.Acquire(
+                    inst.duration, inst.channel, mem_slot=mem_slot))
             # Measurement pulses should only be added if its qubit was measured by the user
```
Comment (lcapelluto) [inline qiskit/pulse/macros.py]:
```suggestion
                inst = instructions.Acquire(inst.duration, inst.channel, mem_slot=mem_slot))
```

File: qiskit/pulse/macros.py [hunk 8]
```diff
@@ -81,11 +81,10 @@ def measure(qubits: List[int],
...
-            elif inst.channels[0].index in qubits:
+            else:
                 schedule = schedule.insert(time, inst)
```
Comment (lcapelluto) [inline qiskit/pulse/macros.py:88]:
```suggestion
            schedule = schedule.insert(time, inst)
```
We can update the inst, and then this is always needed

File: qiskit/pulse/macros.py (hunk 7 above)
Comment (SooluThomas) [reply qiskit/pulse/macros.py]: Fixed in https://github.com/Qiskit/qiskit-terra/pull/5971/commits/1b9ee38a78641265ec75dd672b2101a191aa178f

File: qiskit/pulse/macros.py (hunk 8 above)
Comment (SooluThomas) [reply qiskit/pulse/macros.py]: Fixed in https://github.com/Qiskit/qiskit-terra/pull/5971/commits/1b9ee38a78641265ec75dd672b2101a191aa178f

File: qiskit/assembler/assemble_schedules.py [hunk 9]
```diff
@@ -246,19 +246,28 @@ def _validate_meas_map(instruction_map: Dict[Tuple[int, instructions.Acquire],
...
-        for inst in instrs:
-            measured_qubits.add(inst.channel.index)
-
-        for meas_set in meas_map_sets:
-            intersection = measured_qubits.intersection(meas_set)
-            if intersection and intersection != meas_set:
-                raise QiskitError('Qubits to be acquired: {} do not satisfy required qubits '
-                                  'in measurement map: {}'.format(measured_qubits, meas_set))
+        measured_qubits.update([inst.channel.index for inst in instr[1]],
+                               [inst.channel.index for inst in sorted_inst_map[idx+1][1]])
```
Comment (lcapelluto) [inline qiskit/assembler/assemble_schedules.py:265]: Why do this if there's no overlap?

File: qiskit/assembler/assemble_schedules.py [hunk 10]
```diff
@@ -246,19 +246,28 @@ def _validate_meas_map(instruction_map: Dict[Tuple[int, instructions.Acquire],
...
+                           for inst in insts]
+
+    # 1. if the qubits are not in the meas_map -- Raise Error
+    if not set(all_measured_qubits).issubset(set.union(*meas_map_sets)):
+        raise QiskitError("The measured qubits - {} is not in the measurement "
+                          "map - {}".format(all_measured_qubits, meas_map))
+    # 2. if there is time overlap:
+    #    - if the overlap is in the same meas_map -- Raise Error
+    for idx, instr in enumerate(sorted_inst_map[:-1]):
+        if sorted_inst_map[idx+1][0] == instr[0]:
```
Comment (lcapelluto) [inline qiskit/assembler/assemble_schedules.py:261]: What if the durations are different?

File: qiskit/assembler/assemble_schedules.py [hunk 11]
```diff
@@ -246,19 +246,28 @@ def _validate_meas_map(instruction_map: Dict[Tuple[int, instructions.Acquire],
...
-            measured_qubits.add(inst.channel.index)
-
-        for meas_set in meas_map_sets:
-            intersection = measured_qubits.intersection(meas_set)
-            if intersection and intersection != meas_set:
-                raise QiskitError('Qubits to be acquired: {} do not satisfy required qubits '
-                                  'in measurement map: {}'.format(measured_qubits, meas_set))
+        measured_qubits.update([inst.channel.index for inst in instr[1]],
+                               [inst.channel.index for inst in sorted_inst_map[idx+1][1]])
+        inst_end_time = instr[0][0] + instr[0][1]
```
Comment (lcapelluto) [inline qiskit/assembler/assemble_schedules.py:266]: I'm not sure why this is double indexed -- I feel like it should raise an error. Don't we have this:
`instr = (start_time, duration)`?

File: qiskit/assembler/assemble_schedules.py [hunk 12]
```diff
@@ -246,19 +246,28 @@ def _validate_meas_map(instruction_map: Dict[Tuple[int, instructions.Acquire],
...
+    all_measured_qubits = [inst.channel.index for _, insts in sorted_inst_map
+                           for inst in insts]
+
+    # 1. if the qubits are not in the meas_map -- Raise Error
+    if not set(all_measured_qubits).issubset(set.union(*meas_map_sets)):
+        raise QiskitError("The measured qubits - {} is not in the measurement "
+                          "map - {}".format(all_measured_qubits, meas_map))
+    # 2. if there is time overlap:
+    #    - if the overlap is in the same meas_map -- Raise Error
+    for idx, instr in enumerate(sorted_inst_map[:-1]):
```
Comment (lcapelluto) [inline qiskit/assembler/assemble_schedules.py:260]: `instr` isn't an instruction, it's a tuple with information about timing (`start_time, duration`)

File: qiskit/assembler/assemble_schedules.py (hunk 9 above)
Comment (lcapelluto) [inline qiskit/assembler/assemble_schedules.py:265]: style nitpick:
```suggestion
        measured_qubits = set([inst.channel.index for inst in instr[1]],
                               [inst.channel.index for inst in sorted_inst_map[idx+1][1]])
```

File: qiskit/assembler/assemble_schedules.py [hunk 13]
```diff
@@ -246,19 +246,28 @@ def _validate_meas_map(instruction_map: Dict[Tuple[int, instructions.Acquire],
     Raises:
         QiskitError: If the instructions do not satisfy the measurement map.
     """
+    sorted_inst_map = sorted(instruction_map.items(), key=lambda item: item[0])
```
Comment (lcapelluto) [inline qiskit/assembler/assemble_schedules.py:249]: This doesn't return only keys?

File: qiskit/assembler/assemble_schedules.py [hunk 14]
```diff
@@ -246,19 +246,28 @@ def _validate_meas_map(instruction_map: Dict[Tuple[int, instructions.Acquire],
...
-        for meas_set in meas_map_sets:
-            intersection = measured_qubits.intersection(meas_set)
-            if intersection and intersection != meas_set:
-                raise QiskitError('Qubits to be acquired: {} do not satisfy required qubits '
-                                  'in measurement map: {}'.format(measured_qubits, meas_set))
+        measured_qubits.update([inst.channel.index for inst in instr[1]],
+                               [inst.channel.index for inst in sorted_inst_map[idx+1][1]])
+        inst_end_time = instr[0][0] + instr[0][1]
+        if sorted_inst_map[idx+1][0][0] < inst_end_time and \
+                any(measured_qubits.issubset(meas_set) for meas_set in meas_map_sets):
```
[the remark this answers is not in the document]
Comment (lcapelluto) [reply qiskit/assembler/assemble_schedules.py:268]: I have a suspicion that the integers in the first instruction need to be kept separate from those in the second one

File: qiskit/assembler/assemble_schedules.py (hunk 9 above)
Comment (SooluThomas) [reply qiskit/assembler/assemble_schedules.py]: Fixed in https://github.com/Qiskit/qiskit-terra/pull/5971/commits/03f29b2d740e05351d44cc18185f2e2ef9e413f4

File: test/python/compiler/test_assembler.py [hunk 15]
```diff
@@ -1224,6 +1224,108 @@ def test_assemble_single_instruction(self):
...
+            Acquire(5, AcquireChannel(1), MemorySlot(1)) << 5,
+        )
+        qobj = assemble(schedule,
+                        qubit_lo_freq=self.default_qubit_lo_freq,
+                        meas_lo_freq=self.default_meas_lo_freq,
+                        meas_map=[[0, 1]])
+        validate_qobj_against_schema(qobj)
+
+    def test_assemble_disjoint_time(self):
+        """Test that assembly works when qubits are measured in disjoint time."""
```
Comment (lcapelluto) [inline test/python/compiler/test_assembler.py:1277]:
```suggestion
        """Test that assembly works when qubits are in disjoint meas map sets."""
```

File: test/python/compiler/test_assembler.py [hunk 16]
```diff
@@ -1224,6 +1224,108 @@ def test_assemble_single_instruction(self):
...
+        schedule = schedule.append(
+            Acquire(5, AcquireChannel(1), MemorySlot(1)) << 1,
+        )
+        qobj = assemble(schedule,
+                        qubit_lo_freq=self.default_qubit_lo_freq,
+                        meas_lo_freq=self.default_meas_lo_freq,
+                        meas_map=[[0, 2], [1, 3]])
+        validate_qobj_against_schema(qobj)
+
+    def test_assemble_invalid_qubits(self):
```
Comment (lcapelluto) [inline test/python/compiler/test_assembler.py:1291]: duplicate of `test_assemble_qubits_not_in_meas_map`

Comment (SooluThomas) [reply test/python/compiler/test_assembler.py]: fixed in https://github.com/Qiskit/qiskit-terra/pull/5971/commits/ac87c70236ab323f937d1fa4e4ce78a67ec39e51

**Comment by lcapelluto** [comment]
blocking https://github.com/Qiskit/qiskit-terra/pull/5577

File: releasenotes/notes/update-meas-map-interpretation-53795a9bc7d42857.yaml [hunk 17]
```diff
@@ -0,0 +1,35 @@
+---
+fixes:
+  - |
+    The interpretation of the `meas_map` has historically been that any measurement of
+    `qubit<i>` must be accompanied by a measurement of all qubits in `meas_map[i]`.
+    However, the interpretation should now be that the qubits in `meas_map[i]` may not be
+    measured disjointly in time such that they overlap incompletely.
+    For example:
```
Comment (lcapelluto) [inline releasenotes/notes/update-meas-map-interpretation-53795a9bc7d42857.yaml:8]:
```suggestion
    qubit ``i`` must be accompanied by a measurement of all qubits in the same subgroup as ``i``.
    However, the interpretation should now be that the qubits in that subgrouping may not be
    measured such that they partially overlap.
    For example:
```

File: releasenotes/notes/update-meas-map-interpretation-53795a9bc7d42857.yaml [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+        sched = pulse.Schedule()
+        sched = sched.append(pulse.Acquire(10, acq1))
+        sched = sched.append(pulse.Acquire(10, acq2)) << 10
+
+        meas_map = [[0], [1]]
+        sched = pulse.Schedule()
+        sched = sched.append(pulse.Acquire(10, acq1))
+        sched = sched.append(pulse.Acquire(10, acq2)) << 1
+
+        # Bad
```
Comment (lcapelluto) [inline releasenotes/notes/update-meas-map-interpretation-53795a9bc7d42857.yaml:30]:
```suggestion
        # Good
        meas_map = [[0, 1]]
        # Acquire a subset of [0, 1]
        sched = pulse.Schedule()
        sched = sched.append(pulse.Acquire(10, acq_q0))

        # Acquire 0 and 1 together (same start time, same duration)
        sched = pulse.Schedule()
        sched = sched.append(pulse.Acquire(10, acq_q0))
        sched = sched.append(pulse.Acquire(10, acq_q1))

        # Acquire 0 and 1 disjointly
        sched = pulse.Schedule()
        sched = sched.append(pulse.Acquire(10, acq_q0))
        sched = sched.append(pulse.Acquire(10, acq_q1)) << 10

        # Acquisitions overlap, but 0 and 1 aren't in the same measurement grouping
        meas_map = [[0], [1]]
        sched = pulse.Schedule()
        sched = sched.append(pulse.Acquire(10, acq_q0))
        sched = sched.append(pulse.Acquire(10, acq_q1)) << 1

        # Bad: 0 and 1 are in the same grouping, but acquisitions overlap imperfectly
```

File: releasenotes/notes/update-meas-map-interpretation-53795a9bc7d42857.yaml [hunk 19]
```diff
@@ -0,0 +1,35 @@
...
+        meas_map = [[0, 1]]
+        sched = pulse.Schedule()
+        sched = sched.append(pulse.Acquire(10, acq1))
+        sched = sched.append(pulse.Acquire(10, acq2)) << 1
```
Comment (lcapelluto) [inline releasenotes/notes/update-meas-map-interpretation-53795a9bc7d42857.yaml:34]:
```suggestion
        sched = sched.append(pulse.Acquire(10, acq_q0))
        sched = sched.append(pulse.Acquire(10, acq_q1)) << 1
```

File: qiskit/assembler/assemble_schedules.py [hunk 20]
```diff
@@ -246,20 +246,29 @@ def _validate_meas_map(instruction_map: Dict[Tuple[int, instructions.Acquire],
...
+        if sorted_inst_map[idx+1][0][0] < inst_end_time:
+            instr_qubits = {inst.channel.index for inst in instr[1]}
+            next_instr_qubits = {inst.channel.index for inst in sorted_inst_map[idx+1][1]}
+            for meas_set in meas_map_sets:
+                common_instr_qubits = instr_qubits.intersection(meas_set)
+                common_next = next_instr_qubits.intersection(meas_set)
+                if common_instr_qubits and common_next:
+                    raise QiskitError('Qubits {} and {} in the measurement map: {} was '
+                                      'acquired disjointly'.format(common_instr_qubits,
+                                                                   common_next, meas_map))
```
Comment (lcapelluto) [inline qiskit/assembler/assemble_schedules.py:271]:
```suggestion
                    raise QiskitError('Qubits {} and {} are in the same measurement grouping: {}. '
                                      'They must either be acquired at the same time, or disjointly. '
                                      'Instead, they were acquired at times: {}-{} and {}-{}'.format(common_instr_qubits,
                                                                   common_next, meas_map,
                                                                   instr[0][0], inst_end_time,
                                                                   next_inst_time, next_inst_time + next_inst[0][1]))
```
Going to recommend also adding a couple more variables to make it more readable and to make this suggestion ^ execute

File: qiskit/assembler/assemble_schedules.py [hunk 21]
```diff
@@ -246,20 +246,29 @@ def _validate_meas_map(instruction_map: Dict[Tuple[int, instructions.Acquire],
...
+
+    # 1. if the qubits are not in the meas_map -- Raise Error
+    if not set(all_measured_qubits).issubset(set.union(*meas_map_sets)):
+        raise QiskitError("The measured qubits - {} is not in the measurement "
+                          "map - {}".format(all_measured_qubits, meas_map))
+    # 2. if there is time overlap:
+    #    - if the overlap is in the same meas_map -- Raise Error
+    for idx, instr in enumerate(sorted_inst_map[:-1]):
+        inst_end_time = instr[0][0] + instr[0][1]
+        if sorted_inst_map[idx+1][0][0] < inst_end_time:
```
Comment (lcapelluto) [inline qiskit/assembler/assemble_schedules.py:262]:
```suggestion
        next_instr = sorted_inst_map[idx+1]
        next_instr_time = next_instr[0][0]
        if next_instr_time < inst_end_time:
```
We got a bit inconsistent about abbreviating as `inst` or `instr`. Either is ok -- let's just go with `inst` everywhere?

**Review by lcapelluto — CHANGES_REQUESTED** [review]

File: releasenotes/notes/update-meas-map-interpretation-53795a9bc7d42857.yaml (hunk 18 above)
Comment (SooluThomas) [reply releasenotes/notes/update-meas-map-interpretation-53795a9bc7d42857.yaml]: Fixed in https://github.com/Qiskit/qiskit-terra/pull/5971/commits/e471d666e83904922779f399389fee80600b0fca

File: releasenotes/notes/update-meas-map-interpretation-53795a9bc7d42857.yaml (hunk 19 above)
Comment (SooluThomas) [reply releasenotes/notes/update-meas-map-interpretation-53795a9bc7d42857.yaml]: Fixed in https://github.com/Qiskit/qiskit-terra/pull/5971/commits/e471d666e83904922779f399389fee80600b0fca

File: qiskit/assembler/assemble_schedules.py (hunk 20 above)
Comment (SooluThomas) [reply qiskit/assembler/assemble_schedules.py]: Fixed in https://github.com/Qiskit/qiskit-terra/pull/5971/commits/e471d666e83904922779f399389fee80600b0fca

File: qiskit/assembler/assemble_schedules.py (hunk 21 above)
Comment (SooluThomas) [reply qiskit/assembler/assemble_schedules.py]: Fixed in https://github.com/Qiskit/qiskit-terra/pull/5971/commits/e471d666e83904922779f399389fee80600b0fca and https://github.com/Qiskit/qiskit-terra/pull/5971/commits/22999a40daf444428af9fbd6f7c66b5eef75c5ce

**Review by lcapelluto — COMMENTED** [review]
LGTM!

**Review by lcapelluto — APPROVED** [review]

**Comment by lcapelluto** [comment]
I should try running on a device before adding `automerge`

**Pull request merged**